oak_runtime/src/node/http/server.rs

```diff
@@ -320,43 +310,16 @@ impl HttpRequestHandler {
                 );
 
                 debug!("Injecting the request into the Oak Node");
-                let response = self
-                    .inject_http_request(request, oak_label)
-                    .map_err(|err| {
-                        HttpError::ChannelOperation(format!(
-                            "Error when injecting the request into the Oak Node: {:?}",
-                            err
-                        ))
-                    })?;
+                let response = self.inject_http_request(request, oak_label)?;
 
                 response.try_into_hyper_response()
             }
-            Err(HttpError::RequestLabel(msg)) => {
-                warn!("Invalid or missing Oak label: {}", msg);
-                http::response::Builder::new()
-                    .status(StatusCode::BAD_REQUEST)
-                    .body(Body::from("Invalid or missing Oak label."))
-                    .map_err(|e| {
-                        HttpError::HttpResponse(format!("Could not create response: {}", e))
-                    })
-            }
-            Err(HttpError::IdentityVerification(msg)) => {
-                warn!("Could not verify user's identity: {}", msg);
-                http::response::Builder::new()
-                    .status(StatusCode::BAD_REQUEST)
-                    .body(Body::from("Could not verify user's identity."))
-                    .map_err(|e| {
-                        HttpError::HttpResponse(format!("Could not create response: {}", e))
-                    })
-            }
-            Err(http_err) => {
-                warn!("Internal server error: {:?}", http_err);
+            Err(err) => {
+                warn!("Error when handling the request: {}", err);
```

Comment:
Based on the changes here, I am no longer certain that using anyhow is a good idea, as we cannot distinguish between a bad request and other types of errors. However, maybe the http responses here are not that important after all, since in most cases they are not delivered to the caller. WDYT @tiziano88?

Reply:
What is the effect of this change? That everything will be returned as `StatusCode::INTERNAL_SERVER_ERROR`, even if it is another type of error? If so, that seems indeed too coarse. Or are some more particular errors still being propagated precisely?

Reply:
After a closer look at the code (thanks to [comment](https://github.com/project-oak/oak/issues/1693#issuecomment-759619180) from @ipetr0v), I think it is correct to always send a bad request response, since `get_oak_label` only returns an error if something is wrong with the label. I have changed the code to return `StatusCode::BAD_REQUEST`.